tools/shared.py

```diff
@@ -548,7 +548,7 @@ def check_sanity(force=False):
       logging.critical('The JavaScript shell used for compiling (%s) does not seem to work, check the paths in %s' % (COMPILER_ENGINE, EM_CONFIG))
       sys.exit(1)
 
-    if NODE_JS != COMPILER_ENGINE:
+    if NODE_JS != COMPILER_ENGINE and Settings.RUNNING_JS_OPTS:
```

Comment:
So I realized this doesn't actually work because that settings object isn't set up yet at sanity check time...